apollo-federation/src/connectors/json_selection/lit_expr.rs

```diff
@@ -102,6 +184,13 @@ impl LitExpr {
         }
     }
 
+    fn parse_binary_operator(input: Span) -> ParseResult<LitOp> {
+        alt((
+            map(ranged_span("??"), |_| LitOp::NullCoalescing),
+            map(ranged_span("?!"), |_| LitOp::NoneCoalescing),
```

Comment:
```suggestion
    fn parse_binary_operator(input: Span) -> ParseResult<WithRange<LitOp>> {
        alt((
            map(ranged_span("??"), |qq| WithRange::new(LitOp::NullCoalescing, qq.range())),
            map(ranged_span("?!"), |qi| WithRange::new(LitOp::NoneCoalescing, qi.range())),
```
Unless you want to make `LitOp` carry its own range information somehow, the `WithRange<T>` wrapper is the utility we use around here to preserve source range information.

> [!WARNING]
> Code above hand-written in GitHub web UI, not compiled or tested, and may require other adjustments to work properly.

Reply:
Alternatively, you might just want to implement
```rust
impl LitOp {
    fn parse(input: Span) -> ParseResult<WithRange<Self>> {
        // ... contents of LitExpr::parse_binary_operator here...
    }
}
```